code/datums/actions/xeno_action.dm

```diff
@@ -136,9 +141,11 @@
 	return cooldown_timer
 
 
-/datum/action/xeno_action/proc/add_cooldown()
+/datum/action/xeno_action/proc/add_cooldown(cooldown_modifier = 1, cooldown_override = FALSE)
 	SIGNAL_HANDLER
-	var/cooldown_length = get_cooldown()
+	var/cooldown_length = get_cooldown() * cooldown_modifier
+	if(cooldown_override)
+		cooldown_length = cooldown_override
```

Comment:
why are there two args that do basically the same thing

Reply:
One's a modifier, the other's a straight up override